tools/tsp-client/src/fs.ts

```diff
@@ -13,8 +13,9 @@ export async function removeDirectory(path: string) {
   await rm(path, { recursive: true, force: true });
 }
 
-export async function createTempDirectory(): Promise<string> {
-  const tempRoot = await mkdtemp(path.join(tmpdir(), "tsp-client-"));
+export async function createTempDirectory(outputDir: string): Promise<string> {
```

Comment:
TODO look into combining the helper functions for both the download and github clone scenarios